File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findLHS = function(nums) {
    const map = {};
    let max=0;
    for(const num of nums) 
        map[num] ? map[num]++ : map[num] = 1;
    for(const item in map) 
        if(map[parseInt(item)+1]) 
            max = Math.max(max, map[item]+map[parseInt(item)+1]);
    
    return max;
};

File: 0594-longest-harmonious-subsequence/0594-longest-harmonious-subsequence.java
class Solution {
    public int findLHS(int[] nums) {
        Arrays.sort(nums);
        int max=0, l=0, r=1,n=nums.length;
        while(r<n){
            int diff = nums[r] - nums[l];
            if(diff == 1){
                max = Math.max(max, (r-l+1));
            }
            if(diff <= 1) r++;
            else {
                l++; 
                r=l+1;
            }
        }
        return max;
    }
}